api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -353,11 +334,25 @@ private User initializeUserIfNeeded() {
       request.getDocumentType(), request.getDocumentNumber()
     );
 
-    User user = userProvider.get();
-    user.setBlockscoreId(person.getId());
-    user.setBlockscoreVerificationIsValid(person.isValid());
-    userDao.save(user);
+    User user = userService.setIdVerification(person.getId(), person.isValid());
+    return getProfileResponse(user);
+  }
 
-    return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+  @Override
+  public ResponseEntity<Profile> submitDemographicsSurvey() {
+    User user = userService.submitDemographicSurvey();
```

Comment:
To follow-on to my naming suggestion (below), I would expect a function with the verb "submit" to accept parameters. What am I submitting?

Reply:
See other comments.